### ISSUE TITLE
Add ability to catch Sass @warn messages

### ISSUE THREAD

**Issue by JohnAlbin** [JohnAlbin/sassy-test#1]
Currently, Sassy Test can catch `@error` messages because it causes a full node-sass error object to be generated.

It should be possible to catch `@warn` messages that are sent to standard error.

**Comment by JohnAlbin** [comment]
From https://github.com/sass/node-sass/issues/646#issuecomment-87121010 :

```JavaScript
var sass = require('node-sass');

sass.render({
  data: 'b {l: ah}; @warn "foo";',
  functions: {
    '@warn': function(warning) {
      console.log(warning.getValue());
      return sass.NULL;  // or sass.types.NULL, both work
      // or you can return a regular JS string,
      // or a sass string: return new sass.types.String('squiggly');
      // or elaborated syntax:
      // var str = new sass.types.String;
      // str.setValue('meh');
      //
      // nit: returning an empty JS string "", undefined or null wouldn't work either
    }
  }
}, function(err, result) {
  if(err) {
    console.log(err); throw err;
  }

  console.log(result, result.css.toString());
})
```

**Comment by JohnAlbin** [comment]
Fixed with d4d5cbdc17f890cf8d5ef3b7bdcb98b76151e294
